### DISCUSSION TITLE
Upgrade from v1 to v2 on new server with existing data

### DISCUSSION THREAD

**Discussion by pjv** [postalserver/postal#1605] [Help with using Postal] [answered]
In the new docs there is a section on upgrading from v1 to v2, and within that section there is a subsection titled “Installing on a new server with existing data”.

Within that subsection, step 4 says this:

>4. Use whatever tool takes your fancy (mysqldump, Mariabackup etc...) to copy your databases to your new server. Make sure you copy the postal database as well as all other databases prefixed with postal (or whatever you have configured your prefix to be in the message_db part of your configuration).

But there is no further step that says anything about what to do with the copied db data. How does one import those db dumps into the db on the new server? Is Mariadb supposed to be running on the new server even though step 3 in those instructions says:

>On your new server, stop Postal using postal stop.

I’m familiar with Docker and compose, but confused about the postal v2 architecture. In the repo there is a compose file that includes mariadb and rabbitmq services. Is that part of the standard v2 architecture or an optional way of running Postal v2? I note that in the prerequisites section, a user is guided to spin up mariadba and rabbitmq directly via docker commands which would then be redundant with the services in the compose file.

**Comment by willpower232** [comment, marked as answer]
The instructions presume you are moving all services to a new server, including your databases. Installing every part of Postal and its dependences on the new server is the best way of guaranteeing everything will work on the new server before moving the critical data.

If your databases are on a separate server entirely then you would not have to move them if you don't want to but if you have to move them, you can look up how to move databases as that should be a standard process, not Postal-specific.

If I remember correctly, after using mysqldump, you can execute the output queries in your new database server as they are regular MySQL queries.

The repo's docker-compose file is only for internal testing. The one you use to run Postal is present in the install repo https://github.com/postalserver/install

**Reply by pjv** [reply]
Thanks @willpower232, I was not asking how to restore a mysqldump; I do that every day. 

I guess what I was trying to get at was whether the mariadb server and rabbitmq server are or are not formally considered part of the Postal stack. Your answer indicates to me that they are not; that they should be installed and managed separately from Postal and that the Postal docker install excludes (but depends on) mariadb and rabbitmq.

This is why I asked about whether the DB would still be running after issuing `postal stop` in step 3 of the instructions. If mariadb and rabbitmq were part of the Postal stack, I’d expect that step to shut them down. If not, I’d expect them to stay running.

Personally, leaving mariadb and rabbitmq outside of the formal Postal stack seems like a mistake to me since they are hard dependencies. If they were considered part of a standard Postal install, then the entirety of installing postal could be as simple as spinning up a VPS with docker and docker-compose, creating a directory somewhere convenient, setting up a config file in that directory (which would be bind-mounted into the Postal container), copying a `docker-compose.yml` file (which would include all the necessary Postal service definitions as well as a mariadb service and a rabbitmq service) into the directory, and then issuing `docker-compose up -d`. Since I saw the docker-compose file in the repo, that’s what I was assuming the basic architecture to be.

In that case, the only real artifacts that would need to be supplied to an end-user is the `docker-compose.yml` file (which would specify ALL services needed to run Postal) and some documentation about how to do whatever configuration is needed.

**Reply by willpower232** [reply]
Thanks for the clarification.

Yes they are hard dependencies in that you have to use those two pieces of software, or at least software fully compatible on those protocols if any others exist, however there are so many different and valid ways of configuring and managing them that it would be impossible to fully support both them and Postal.

In this case, you would need MariaDB running otherwise you would not be able to import the databases in the usual way and it is safer if Postal is not also trying to make use of the databases whilst they're being restored.

The existence of the official Docker image also makes any advanced way of running Postal possible, such as kubernetes or other cluster-like environment, without complicating the support since the Docker image should always be the same so now there is a middle ground in running Postal however you like and being able to get support for it.
